### DISCUSSION TITLE
Toggle View Interactivity

### DISCUSSION THREAD

**Discussion by mcost45** [visgl/deck.gl#5839] [Q&A]
Hi, just wondering if there are any best practices for toggling the interactivity of a view?

I have an interactive minimap view (with a controller) that allows for navigating the main view. I am aiming to be able to toggle this minimp on and off. It is easy enough to toggle the visibility of the layers from the minimap so it can no longer be seen, but the view area itself will still retain the same minimap interactivity when clicked on.

I've tried a few different methods like changing deck's view array prop/attempting to map dynamic view state changes to act as if they were from the main view, but these have not yet been fully successful. Does anyone have any suggestions?

Thank you in advance!

**Comment by Pessimistress** [comment]
> I've tried a few different methods like changing deck's view array prop/attempting to map dynamic view state changes to act as if they were from the main view, but these have not yet been fully successful.

Can you share the code of what you have tried? Removing a view from the `views` prop should be sufficient.

**Reply by mcost45** [reply]
@Pessimistress Removing the view does work for removing interactivity, but my problem is then being able to toggle the minimap back on, by adding the view back to the views prop. After I add the minimap view back, interactions on the minimap will instead 'fall-through' to the main view underneath instead (i.e. panning etc) - though before it was hidden it would work properly. I've verified visually with a clear colour that the interactive minimap view does get removed and added again, above the main view - but I am not sure why the controller no longer works properly.

Here's the most relevant code:

- When toggling the view based on the boolean 'visible' (views is an array of view objects with said minimap view stored at [2]):

```
if (visible) {
    // create a new minimap view object (same function responsible for its original, working creation)
    // with controller: true
    views[2] = getNewDynamicMinimapView();
    // I've tested resetting its viewState back to its initial state as well, but this doesn't seem to change anything
    viewStates.dynamicMinimapView = initialViewStates.dynamicMinimapView;
} else {
    // remove the minimap view from the views array
    views[2] = null;
}
deck.setProps({
    viewState: {
        dynamicMinimapView: {...viewStates.dynamicMinimapView}
    },
    views: [...views]
});
```

- Get new minimap view object function:
```
getNewDynamicMinimapView() {
    return new OrthographicView(
        {id: dynamicViewMinimapView, controller: MinimapController}
    );
}
```

MinimapController is an extension of OrthographicController, it is the exact same except it also has additional 'mousedown' event functionality. With my implementation, _after_ I toggle the minimap view off then back on again, deck will pickup the initial 'mousedown' event on the minimap view via onViewStateChange, but then the following pan events from dragging will be triggered from the main view, even though the click began on the minimap view. _Before_ I toggle the minimap off then back on again (initially), clicking and dragging on the minimap view will trigger the panning events from the minimap view as expected, not from the main view.

Is this sufficient information? And thank you!

**Reply by Pessimistress** [reply]
Thank you! Could you open a bug with the above write up?
